wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth1/OAuthAuthenticator.java

```diff
@@ -229,6 +229,14 @@ String callback(final URL requestUrl) throws OAuthAuthenticationException {
    */
   abstract String getOAuthProvider();
 
+  /**
+   * Returns URL to initiate authentication process using given authenticator. Typically points to
+   * {@code /api/oauth/} or {@code /api/oauth/1.0} endpoint with necessary request params.
+   *
+   * @return URL to initiate authentication process
+   */
+  public abstract String getLocalAuthenticateUrl();
```

Comment:
what was the motivation to put `local` in the name? What about `getAuthenticateUrl`?

Reply:
there is already one for remote URL-s
https://github.com/eclipse/che/blob/108be5cd84b8535e38f3ac500e36e1fbc2e4f4e6/wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth1/OAuthAuthenticator.java#L102-L106

Reply:
ok. and? 
What exactly means local?

Reply:
local means that this URL will point to the local authettication REST method, not the remote server auth URL.

Reply:
ok